processor/unrollprocessor/processor.go

```diff
@@ -0,0 +1,114 @@
+package unrollprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type unrollProcessor struct {
+	cfg *Config
+}
+
+// newUnrollProcessor returns a new unrollProcessor.
+func newUnrollProcessor(config *Config) (*unrollProcessor, error) {
+	if err := config.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return &unrollProcessor{
+		cfg: config,
+	}, nil
+}
+
+// ProcessLogs implements the processor interface
+func (p *unrollProcessor) ProcessLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	newLogs := plog.NewLogs()
```

Comment:
Why do we need to create a new one? Can't we just move log records around within each `LogRecordSlice`?